From e2064b1952b4b3717c2e1dc171c27b959e268629 Mon Sep 17 00:00:00 2001
From: Or Magen <ormagen@amazon.com>
Date: Mon, 19 Jan 2026 16:36:21 +0200
Subject: [PATCH 3/5] test: Verify pahole extractor completeness on real data

Validated on al_report.o with 228 structs:
- 227 structs parsed (1 built-in skipped)
- 202 unique structs after deduplication
- al_report: 20/21 members (vtable correctly skipped)
- All 15 edge case tests pass
- Full test suite: 425 tests pass
---
 patches/APPLY_ORDER.txt                       |  2 +-
 patches/struct_000_A.patch                    |  6 ++--
 patches/struct_000_AfterRegion.patch          |  6 ++--
 patches/struct_000_Arrays.patch               |  6 ++--
 patches/struct_000_Base.patch                 |  6 ++--
 patches/struct_000_Bits.patch                 |  6 ++--
 patches/struct_000_BlockCommentIgnored.patch  |  6 ++--
 .../struct_000_BlockWhitespaceIgnored.patch   |  6 ++--
 patches/struct_000_Complex.patch              |  9 ++++++
 patches/struct_000_Config.msg                 |  4 +--
 patches/struct_000_Config.patch               | 12 ++++---
 patches/struct_000_Data.msg                   |  4 +--
 patches/struct_000_Data.patch                 | 31 +++++++++----------
 patches/struct_000_FirstAfter.patch           |  6 ++--
 patches/struct_000_FirstIgnored.patch         |  6 ++--
 patches/struct_000_FirstLock.patch            | 14 ++++-----
 patches/struct_000_FirstStruct.patch          |  6 ++--
 patches/struct_000_Free.patch                 |  6 ++--
 patches/struct_000_IgnoredStruct.msg          |  4 +--
 patches/struct_000_IgnoredStruct.patch        | 28 +++++++++--------
 patches/struct_000_Inner.patch                |  6 ++--
 patches/struct_000_Large.patch                |  6 ++--
 patches/struct_000_LastLock.patch             |  6 ++--
 patches/struct_000_Locked.patch               | 10 +++---
 patches/struct_000_LockedMember.patch         | 17 +++++-----
 patches/struct_000_MiddleLock.patch           | 13 ++++----
 patches/struct_000_Mixed.patch                |  6 ++--
 patches/struct_000_MixedCaseIgnored.patch     |  6 ++--
 patches/struct_000_Multi.patch                |  9 ++++++
 patches/struct_000_MultipleLocks.patch        |  6 ++--
 patches/struct_000_Optimal.patch              |  6 ++--
 patches/struct_000_Outer.patch                | 21 +++++++++++++
 patches/struct_000_Partial.patch              | 10 +++---
 patches/struct_000_Ptrs.patch                 |  6 ++--
 patches/struct_000_S.patch                    |  6 ++--
 patches/struct_000_SingleLock.patch           |  6 ++--
 patches/struct_001_B.patch                    |  6 ++--
 patches/struct_001_BeforeRegion.patch         |  6 ++--
 patches/struct_001_Derived.patch              | 16 ----------
 patches/struct_001_FirstBefore.patch          |  6 ++--
 patches/struct_001_NormalStruct.patch         |  6 ++--
 patches/struct_001_Outer.patch                |  6 ++--
 patches/struct_001_ProcessedStruct.patch      |  6 ++--
 patches/struct_001_SecondStruct.patch         |  6 ++--
 patches/struct_002_FirstInRegion.patch        |  6 ++--
 patches/struct_002_InOffRegion.patch          |  6 ++--
 patches/struct_002_InRegion.patch             |  6 ++--
 patches/struct_002_InnerRegion.patch          |  6 ++--
 patches/struct_002_SecondIgnored.patch        |  6 ++--
 patches/struct_002_SingleLineIgnored.patch    |  6 ++--
 patches/struct_002_UpperCaseIgnored.patch     |  6 ++--
 patches/struct_002_WhitespaceIgnored.patch    |  6 ++--
 patches/struct_003_OuterRegion.patch          |  6 ++--
 patches/struct_003_SecondAfter.patch          |  6 ++--
 patches/struct_003_SecondInRegion.patch       |  6 ++--
 patches/struct_004_SecondBefore.patch         |  6 ++--
 patches/struct_004_StillInOuter.patch         |  6 ++--
 57 files changed, 238 insertions(+), 212 deletions(-)

diff --git a/patches/APPLY_ORDER.txt b/patches/APPLY_ORDER.txt
index 3282802..3ff6fbf 100644
--- a/patches/APPLY_ORDER.txt
+++ b/patches/APPLY_ORDER.txt
@@ -1,3 +1,3 @@
 # Apply patches in this order:
 
-1. struct_000_Outer.patch
+1. struct_000_Optimal.patch
diff --git a/patches/struct_000_A.patch b/patches/struct_000_A.patch
index 3ab35cb..cd6f4a9 100644
--- a/patches/struct_000_A.patch
+++ b/patches/struct_000_A.patch
@@ -1,7 +1,7 @@
-diff --git var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpqzajsmyo.cpp var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp78bpglux.cpp
+diff --git tmp/tmp47q570ti.cpp tmp/tmpi64tcpm2.cpp
 index dd6c09b..93be954 100644
---- var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpqzajsmyo.cpp
-+++ var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp78bpglux.cpp
+--- tmp/tmp47q570ti.cpp
++++ tmp/tmpi64tcpm2.cpp
 @@ -1 +1 @@
 -struct A { char x; int y; }; int main() { A a; return 0; }
 \ No newline at end of file
diff --git a/patches/struct_000_AfterRegion.patch b/patches/struct_000_AfterRegion.patch
index d911eaa..b4907a4 100644
--- a/patches/struct_000_AfterRegion.patch
+++ b/patches/struct_000_AfterRegion.patch
@@ -1,7 +1,7 @@
-diff --git var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp5c94d3xv.cpp var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp9w_c_w32.cpp
+diff --git tmp/tmps8dhen7j.cpp tmp/tmpmyi9kiv3.cpp
 index 8bc95d4..3c6c23d 100644
---- var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp5c94d3xv.cpp
-+++ var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp9w_c_w32.cpp
+--- tmp/tmps8dhen7j.cpp
++++ tmp/tmpmyi9kiv3.cpp
 @@ -28,8 +28,8 @@ struct StillInOuter {
  // paddington-on
  
diff --git a/patches/struct_000_Arrays.patch b/patches/struct_000_Arrays.patch
index 28869af..118b28e 100644
--- a/patches/struct_000_Arrays.patch
+++ b/patches/struct_000_Arrays.patch
@@ -1,7 +1,7 @@
-diff --git var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpkx4v758_.cpp var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp71w2wu8_.cpp
+diff --git tmp/tmpx0otvddq.cpp tmp/tmps69osmq6.cpp
 index 617a21b..09ec0d3 100644
---- var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpkx4v758_.cpp
-+++ var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp71w2wu8_.cpp
+--- tmp/tmpx0otvddq.cpp
++++ tmp/tmps69osmq6.cpp
 @@ -1 +1 @@
 -struct Arrays { char a; int b[2]; char c; int d[2]; }; int main() { Arrays arr; return 0; }
 \ No newline at end of file
diff --git a/patches/struct_000_Base.patch b/patches/struct_000_Base.patch
index a42eec3..dfaca08 100644
--- a/patches/struct_000_Base.patch
+++ b/patches/struct_000_Base.patch
@@ -1,7 +1,7 @@
-diff --git var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp0ai3uluv.cpp var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpfvgf0c5d.cpp
+diff --git tmp/tmpxu9fhet5.cpp tmp/tmpmkmgl352.cpp
 index 5983060..089f48b 100644
---- var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp0ai3uluv.cpp
-+++ var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpfvgf0c5d.cpp
+--- tmp/tmpxu9fhet5.cpp
++++ tmp/tmpmkmgl352.cpp
 @@ -1,8 +1,8 @@
  
              struct Base {
diff --git a/patches/struct_000_Bits.patch b/patches/struct_000_Bits.patch
index eced38f..fc606b8 100644
--- a/patches/struct_000_Bits.patch
+++ b/patches/struct_000_Bits.patch
@@ -1,7 +1,7 @@
-diff --git var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmprckrtyno.cpp var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmplcysu_6i.cpp
+diff --git tmp/tmpjsympgkl.cpp tmp/tmpn_8spmmp.cpp
 index ab5a36f..2679059 100644
---- var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmprckrtyno.cpp
-+++ var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmplcysu_6i.cpp
+--- tmp/tmpjsympgkl.cpp
++++ tmp/tmpn_8spmmp.cpp
 @@ -1 +1 @@
 -struct Bits { char a; int b : 4; int c : 4; char d; }; int main() { Bits b; return 0; }
 \ No newline at end of file
diff --git a/patches/struct_000_BlockCommentIgnored.patch b/patches/struct_000_BlockCommentIgnored.patch
index 0aa055f..aeeae88 100644
--- a/patches/struct_000_BlockCommentIgnored.patch
+++ b/patches/struct_000_BlockCommentIgnored.patch
@@ -1,7 +1,7 @@
-diff --git var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp34xewo5g.cpp var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpqz90idat.cpp
+diff --git tmp/tmpa_opfllw.cpp tmp/tmphsnmaugl.cpp
 index 4c04dff..b1ad742 100644
---- var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp34xewo5g.cpp
-+++ var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpqz90idat.cpp
+--- tmp/tmpa_opfllw.cpp
++++ tmp/tmphsnmaugl.cpp
 @@ -6,9 +6,9 @@
              };
              /* paddington-ignore */
diff --git a/patches/struct_000_BlockWhitespaceIgnored.patch b/patches/struct_000_BlockWhitespaceIgnored.patch
index 79b5a62..fe268eb 100644
--- a/patches/struct_000_BlockWhitespaceIgnored.patch
+++ b/patches/struct_000_BlockWhitespaceIgnored.patch
@@ -1,7 +1,7 @@
-diff --git var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp638rzk6i.cpp var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpziierijf.cpp
+diff --git tmp/tmpr51melex.cpp tmp/tmp2bn0v884.cpp
 index 420b9bb..b8acc48 100644
---- var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp638rzk6i.cpp
-+++ var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpziierijf.cpp
+--- tmp/tmpr51melex.cpp
++++ tmp/tmp2bn0v884.cpp
 @@ -6,9 +6,9 @@
              };
              /*  paddington-ignore  */
diff --git a/patches/struct_000_Complex.patch b/patches/struct_000_Complex.patch
index e69de29..682ceb4 100644
--- a/patches/struct_000_Complex.patch
+++ b/patches/struct_000_Complex.patch
@@ -0,0 +1,9 @@
+diff --git tmp/tmpudqlb680.cpp tmp/tmpd4ruvrnk.cpp
+index 9f3252b..b4fe69d 100644
+--- tmp/tmpudqlb680.cpp
++++ tmp/tmpd4ruvrnk.cpp
+@@ -1 +1 @@
+-struct Complex { int width; int height; int area; int* data; Complex(int w, int h) : width(w), height(h), area(width * height), data(new int[area]) {} }; int main() { Complex c(10, 20); return 0; }
+\ No newline at end of file
++struct Complex { int* data; int width; int height; int area; Complex(int w, int h) : width(w), height(h), area(width * height), data(new int[area]) {} }; int main() { Complex c(10, 20); return 0; }
+\ No newline at end of file
diff --git a/patches/struct_000_Config.msg b/patches/struct_000_Config.msg
index 3b48c5b..dea3f6b 100644
--- a/patches/struct_000_Config.msg
+++ b/patches/struct_000_Config.msg
@@ -1,7 +1,7 @@
 refactor: Optimize padding for Config
 
 Reorder members from largest to smallest.
-Saves 8 bytes per instance.
+Saves 0 bytes per instance.
 
 Before: 48 bytes
-After: 40 bytes
\ No newline at end of file
+After: 48 bytes
\ No newline at end of file
diff --git a/patches/struct_000_Config.patch b/patches/struct_000_Config.patch
index 3294554..d0a5f50 100644
--- a/patches/struct_000_Config.patch
+++ b/patches/struct_000_Config.patch
@@ -1,8 +1,8 @@
-diff --git var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp_z079jbr.cpp var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp6xx2l4bb.cpp
-index 323a733..47d384c 100644
---- var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp_z079jbr.cpp
-+++ var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp6xx2l4bb.cpp
-@@ -1,11 +1,9 @@
+diff --git tmp/tmpmq0vkazy.cpp tmp/tmpwz35drpd.cpp
+index 323a733..b6fc21a 100644
+--- tmp/tmpmq0vkazy.cpp
++++ tmp/tmpwz35drpd.cpp
+@@ -1,11 +1,11 @@
  
  struct Config {
 -    char mode;
@@ -12,6 +12,8 @@ index 323a733..47d384c 100644
      short id;
 -};
 +char mode;
++    static int count;
++    static const char* name;
 +    };
  
  int Config::count = 0;
diff --git a/patches/struct_000_Data.msg b/patches/struct_000_Data.msg
index 3be8a2e..d09596e 100644
--- a/patches/struct_000_Data.msg
+++ b/patches/struct_000_Data.msg
@@ -3,5 +3,5 @@ refactor: Optimize padding for Data
 Reorder members from largest to smallest.
 Saves 0 bytes per instance.
 
-Before: 48 bytes
-After: 48 bytes
\ No newline at end of file
+Before: 44 bytes
+After: 44 bytes
\ No newline at end of file
diff --git a/patches/struct_000_Data.patch b/patches/struct_000_Data.patch
index 11fa18e..cdc5fba 100644
--- a/patches/struct_000_Data.patch
+++ b/patches/struct_000_Data.patch
@@ -1,17 +1,16 @@
-diff --git var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmps26kz_cw.cpp var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpuw9_r_cr.cpp
-index 1ab7210..c4fbdc3 100644
---- var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmps26kz_cw.cpp
-+++ var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpuw9_r_cr.cpp
-@@ -1,9 +1,9 @@
+diff --git tmp/tmp5qyibkj8.cpp tmp/tmps_nqbjjr.cpp
+index 815d49e..d17315e 100644
+--- tmp/tmp5qyibkj8.cpp
++++ tmp/tmps_nqbjjr.cpp
+@@ -1,8 +1,8 @@
  
- struct Data {
--    char a;
--    friend void access(Data& d);
-     int b;
--};
-+char a;
-+    friend void access(Data& d);
-+    };
- 
- void access(Data& d) {
-     d.a = 'x';
+             struct Data {
+-                char a;  // first member
+-                int b;   // second member
+-                char c;  // third member
++                int b;   char a;  char c;  // first member
++                // second member
++                // third member
+             };
+             int main() {
+                 Data d;  // instance
diff --git a/patches/struct_000_FirstAfter.patch b/patches/struct_000_FirstAfter.patch
index 7ed7c48..c5356cc 100644
--- a/patches/struct_000_FirstAfter.patch
+++ b/patches/struct_000_FirstAfter.patch
@@ -1,7 +1,7 @@
-diff --git var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpid0bvq3u.cpp var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpzmbe76uj.cpp
+diff --git tmp/tmpstnxwd6n.cpp tmp/tmp2btt813_.cpp
 index 24427e0..38aab98 100644
---- var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpid0bvq3u.cpp
-+++ var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpzmbe76uj.cpp
+--- tmp/tmpstnxwd6n.cpp
++++ tmp/tmp2btt813_.cpp
 @@ -20,8 +20,8 @@ struct InRegion {
  // paddington-on
  
diff --git a/patches/struct_000_FirstIgnored.patch b/patches/struct_000_FirstIgnored.patch
index ab64516..c6c6d77 100644
--- a/patches/struct_000_FirstIgnored.patch
+++ b/patches/struct_000_FirstIgnored.patch
@@ -1,7 +1,7 @@
-diff --git var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp59nd8t3e.cpp var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp9ftpqeqy.cpp
+diff --git tmp/tmpnaj04r8g.cpp tmp/tmpnc06h5y4.cpp
 index 1d95166..ebe12f2 100644
---- var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp59nd8t3e.cpp
-+++ var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp9ftpqeqy.cpp
+--- tmp/tmpnaj04r8g.cpp
++++ tmp/tmpnc06h5y4.cpp
 @@ -1,9 +1,9 @@
  
              // paddington-ignore
diff --git a/patches/struct_000_FirstLock.patch b/patches/struct_000_FirstLock.patch
index 1bbb4fe..9299fe8 100644
--- a/patches/struct_000_FirstLock.patch
+++ b/patches/struct_000_FirstLock.patch
@@ -1,14 +1,14 @@
-diff --git var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmphxxwq4e6.cpp var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpqxyfifwf.cpp
-index 920b9be..9237298 100644
---- var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmphxxwq4e6.cpp
-+++ var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpqxyfifwf.cpp
+diff --git tmp/tmpesm1f3rr.cpp tmp/tmp_qgkv3pa.cpp
+index 920b9be..0d3020d 100644
+--- tmp/tmpesm1f3rr.cpp
++++ tmp/tmp_qgkv3pa.cpp
 @@ -1,10 +1,10 @@
  
  struct FirstLock {
 -    int locked_first; // paddington-lock
-+    int locked_first; int large[50];
-     char small1;
--    int large[50];
+-    char small1;
+     int large[50];
++    int locked_first; char small1;
      char small2;
 -};
 +// paddington-lock
diff --git a/patches/struct_000_FirstStruct.patch b/patches/struct_000_FirstStruct.patch
index 92cff4d..a088603 100644
--- a/patches/struct_000_FirstStruct.patch
+++ b/patches/struct_000_FirstStruct.patch
@@ -1,7 +1,7 @@
-diff --git var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpnm3n1p69.cpp var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpi5zoxmo3.cpp
+diff --git tmp/tmpgcj__bks.cpp tmp/tmp3sf7pc6e.cpp
 index cb6de31..cfb3afb 100644
---- var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpnm3n1p69.cpp
-+++ var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpi5zoxmo3.cpp
+--- tmp/tmpgcj__bks.cpp
++++ tmp/tmp3sf7pc6e.cpp
 @@ -1,8 +1,8 @@
  
  // paddington-off
diff --git a/patches/struct_000_Free.patch b/patches/struct_000_Free.patch
index d98d444..bedc55a 100644
--- a/patches/struct_000_Free.patch
+++ b/patches/struct_000_Free.patch
@@ -1,7 +1,7 @@
-diff --git var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmptccgta8p.cpp var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpqk6efq_b.cpp
+diff --git tmp/tmpsst6zv4g.cpp tmp/tmp4nxd5pmq.cpp
 index 81eaf96..47a6d17 100644
---- var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmptccgta8p.cpp
-+++ var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpqk6efq_b.cpp
+--- tmp/tmpsst6zv4g.cpp
++++ tmp/tmp4nxd5pmq.cpp
 @@ -1 +1 @@
 -struct Free { char a; int b; char c; Free(int x, char y, char z) : a(y), b(x), c(z) {} }; int main() { Free f(42, 'x', 'y'); return 0; }
 \ No newline at end of file
diff --git a/patches/struct_000_IgnoredStruct.msg b/patches/struct_000_IgnoredStruct.msg
index 374938f..833ebeb 100644
--- a/patches/struct_000_IgnoredStruct.msg
+++ b/patches/struct_000_IgnoredStruct.msg
@@ -3,5 +3,5 @@ refactor: Optimize padding for IgnoredStruct
 Reorder members from largest to smallest.
 Saves 0 bytes per instance.
 
-Before: 80 bytes
-After: 80 bytes
\ No newline at end of file
+Before: 64 bytes
+After: 64 bytes
\ No newline at end of file
diff --git a/patches/struct_000_IgnoredStruct.patch b/patches/struct_000_IgnoredStruct.patch
index 3f1da4a..dd96f5b 100644
--- a/patches/struct_000_IgnoredStruct.patch
+++ b/patches/struct_000_IgnoredStruct.patch
@@ -1,14 +1,16 @@
-diff --git var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpa8km8a5i.cpp var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpe5zuon71.cpp
-index 6d0f0b6..a4bd83e 100644
---- var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpa8km8a5i.cpp
-+++ var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpe5zuon71.cpp
-@@ -1,8 +1,8 @@
- 
- // paddington-ignore
- struct IgnoredStruct {
--    char small1;
-     int large[100];
-+    char small1;
-     char small2;
- };
+diff --git tmp/tmpmvs3u0zg.cpp tmp/tmpfnux5ymm.cpp
+index ace835b..15c823f 100644
+--- tmp/tmpmvs3u0zg.cpp
++++ tmp/tmpfnux5ymm.cpp
+@@ -1,9 +1,9 @@
  
+             // paddington-ignore
+             struct IgnoredStruct {
+-                char a;
+                 int b;
+-            };
++            char a;
++                };
+             struct ProcessedStruct {
+                 char a;
+                 int b;
diff --git a/patches/struct_000_Inner.patch b/patches/struct_000_Inner.patch
index de7ffd3..444fbf4 100644
--- a/patches/struct_000_Inner.patch
+++ b/patches/struct_000_Inner.patch
@@ -1,7 +1,7 @@
-diff --git var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp1cdag8k1.cpp var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpfa896y_6.cpp
+diff --git tmp/tmpmscmo3ax.cpp tmp/tmpvj9s0fhs.cpp
 index b83699b..4607e8d 100644
---- var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp1cdag8k1.cpp
-+++ var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpfa896y_6.cpp
+--- tmp/tmpmscmo3ax.cpp
++++ tmp/tmpvj9s0fhs.cpp
 @@ -1 +1 @@
 -struct Inner { char x; int y; }; struct Outer { char a; Inner b; char c; }; int main() { Outer o; return 0; }
 \ No newline at end of file
diff --git a/patches/struct_000_Large.patch b/patches/struct_000_Large.patch
index 28c69e0..aa4fe20 100644
--- a/patches/struct_000_Large.patch
+++ b/patches/struct_000_Large.patch
@@ -1,7 +1,7 @@
-diff --git var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp4sey82nj.cpp var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpgksmeaqy.cpp
+diff --git tmp/tmpkk8eqb66.cpp tmp/tmp6aun5v9u.cpp
 index 8bc325c..3ed0397 100644
---- var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp4sey82nj.cpp
-+++ var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpgksmeaqy.cpp
+--- tmp/tmpkk8eqb66.cpp
++++ tmp/tmp6aun5v9u.cpp
 @@ -1 +1 @@
 -struct Large { char a; long b; char c; long d; char e; long f; }; int main() { Large l; return 0; }
 \ No newline at end of file
diff --git a/patches/struct_000_LastLock.patch b/patches/struct_000_LastLock.patch
index 42e10e0..1e0c414 100644
--- a/patches/struct_000_LastLock.patch
+++ b/patches/struct_000_LastLock.patch
@@ -1,7 +1,7 @@
-diff --git var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp5i7_tl90.cpp var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp7fkj3txa.cpp
+diff --git tmp/tmpwllf_hby.cpp tmp/tmpa0hh40oi.cpp
 index a2c9b15..2b371b1 100644
---- var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp5i7_tl90.cpp
-+++ var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp7fkj3txa.cpp
+--- tmp/tmpwllf_hby.cpp
++++ tmp/tmpa0hh40oi.cpp
 @@ -1,9 +1,9 @@
  
  struct LastLock {
diff --git a/patches/struct_000_Locked.patch b/patches/struct_000_Locked.patch
index 7ab0e2e..046793a 100644
--- a/patches/struct_000_Locked.patch
+++ b/patches/struct_000_Locked.patch
@@ -1,9 +1,9 @@
-diff --git var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpvno5bmfn.cpp var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpblgw_hnm.cpp
-index dee2635..aa2b485 100644
---- var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpvno5bmfn.cpp
-+++ var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpblgw_hnm.cpp
+diff --git tmp/tmpfspoybha.cpp tmp/tmp35bmq68v.cpp
+index dee2635..924a40d 100644
+--- tmp/tmpfspoybha.cpp
++++ tmp/tmp35bmq68v.cpp
 @@ -1 +1 @@
 -struct Locked { char flag; int size; char* buffer; Locked(int s) : flag(1), size(s), buffer(size > 0 ? new char[size] : nullptr) {} }; int main() { Locked l(100); return 0; }
 \ No newline at end of file
-+struct Locked { int size; char* buffer; char flag; Locked(int s) : flag(1), size(s), buffer(size > 0 ? new char[size] : nullptr) {} }; int main() { Locked l(100); return 0; }
++struct Locked { char* buffer; int size; char flag; Locked(int s) : flag(1), size(s), buffer(size > 0 ? new char[size] : nullptr) {} }; int main() { Locked l(100); return 0; }
 \ No newline at end of file
diff --git a/patches/struct_000_LockedMember.patch b/patches/struct_000_LockedMember.patch
index 0b23918..7770b79 100644
--- a/patches/struct_000_LockedMember.patch
+++ b/patches/struct_000_LockedMember.patch
@@ -1,15 +1,16 @@
-diff --git var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp9qvr2xae.cpp var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp66eb6zlg.cpp
-index e6eb78f..842d254 100644
---- var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp9qvr2xae.cpp
-+++ var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp66eb6zlg.cpp
+diff --git tmp/tmp7426bw1u.cpp tmp/tmpb3h1bfbr.cpp
+index e6eb78f..086208d 100644
+--- tmp/tmp7426bw1u.cpp
++++ tmp/tmpb3h1bfbr.cpp
 @@ -1,11 +1,11 @@
  
  struct LockedMember {
-+    int locked_field; int large[50];
-     char small1;
+-    char small1;
 -    int locked_field; // paddington-lock
-     char small2;
--    int large[50];
+-    char small2;
+     int large[50];
++    int locked_field; char small1;
++    char small2;
      char small3;
 -};
 +// paddington-lock
diff --git a/patches/struct_000_MiddleLock.patch b/patches/struct_000_MiddleLock.patch
index f6d2768..d58072d 100644
--- a/patches/struct_000_MiddleLock.patch
+++ b/patches/struct_000_MiddleLock.patch
@@ -1,16 +1,15 @@
-diff --git var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpehhmrtm_.cpp var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpsoz4lonw.cpp
-index ded2a44..f851253 100644
---- var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpehhmrtm_.cpp
-+++ var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpsoz4lonw.cpp
+diff --git tmp/tmpugwfb3o6.cpp tmp/tmpuno7m3hg.cpp
+index ded2a44..e3a44bf 100644
+--- tmp/tmpugwfb3o6.cpp
++++ tmp/tmpuno7m3hg.cpp
 @@ -1,11 +1,11 @@
  
  struct MiddleLock {
 -    char small1;
      int large1[30];
 -    int locked_middle; // paddington-lock
--    int large2[30];
-+    int locked_middle; int large2[30];
-+    char small1;
+     int large2[30];
++    int locked_middle; char small1;
      char small2;
 -};
 +// paddington-lock
diff --git a/patches/struct_000_Mixed.patch b/patches/struct_000_Mixed.patch
index 21bf6b1..82aa071 100644
--- a/patches/struct_000_Mixed.patch
+++ b/patches/struct_000_Mixed.patch
@@ -1,7 +1,7 @@
-diff --git var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpl0uyi3n5.cpp var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp0l8k8__k.cpp
+diff --git tmp/tmp4g03z8lu.cpp tmp/tmpyybux4t6.cpp
 index 2b041ca..ba5788b 100644
---- var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpl0uyi3n5.cpp
-+++ var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp0l8k8__k.cpp
+--- tmp/tmp4g03z8lu.cpp
++++ tmp/tmpyybux4t6.cpp
 @@ -1 +1 @@
 -struct Mixed { char a; double b; short c; int d; }; int main() { Mixed m; return 0; }
 \ No newline at end of file
diff --git a/patches/struct_000_MixedCaseIgnored.patch b/patches/struct_000_MixedCaseIgnored.patch
index 2e056cb..e34e964 100644
--- a/patches/struct_000_MixedCaseIgnored.patch
+++ b/patches/struct_000_MixedCaseIgnored.patch
@@ -1,7 +1,7 @@
-diff --git var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp0oxjbs2r.cpp var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpblkzgsly.cpp
+diff --git tmp/tmpk2vy5axl.cpp tmp/tmpy_8032sd.cpp
 index 742238f..fb89f71 100644
---- var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp0oxjbs2r.cpp
-+++ var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpblkzgsly.cpp
+--- tmp/tmpk2vy5axl.cpp
++++ tmp/tmpy_8032sd.cpp
 @@ -6,9 +6,9 @@
              };
              // Paddington-Ignore
diff --git a/patches/struct_000_Multi.patch b/patches/struct_000_Multi.patch
index e69de29..aa1901c 100644
--- a/patches/struct_000_Multi.patch
+++ b/patches/struct_000_Multi.patch
@@ -0,0 +1,9 @@
+diff --git tmp/tmpcx4_q0on.cpp tmp/tmp7z8_9mld.cpp
+index 8f27cde..99c67e8 100644
+--- tmp/tmpcx4_q0on.cpp
++++ tmp/tmp7z8_9mld.cpp
+@@ -1 +1 @@
+-struct Multi { int width; int height; int* data; Multi(int w, int h) : width(w), height(h), data(new int[width * height]) {} }; int main() { Multi m(5, 10); return 0; }
+\ No newline at end of file
++struct Multi { int* data; int width; int height; Multi(int w, int h) : width(w), height(h), data(new int[width * height]) {} }; int main() { Multi m(5, 10); return 0; }
+\ No newline at end of file
diff --git a/patches/struct_000_MultipleLocks.patch b/patches/struct_000_MultipleLocks.patch
index 5597086..d201f70 100644
--- a/patches/struct_000_MultipleLocks.patch
+++ b/patches/struct_000_MultipleLocks.patch
@@ -1,7 +1,7 @@
-diff --git var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp52lbilra.cpp var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpmkm5bnmu.cpp
+diff --git tmp/tmpcl8bvyhf.cpp tmp/tmpsw5b6fkz.cpp
 index eb4efc9..2acc498 100644
---- var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp52lbilra.cpp
-+++ var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpmkm5bnmu.cpp
+--- tmp/tmpcl8bvyhf.cpp
++++ tmp/tmpsw5b6fkz.cpp
 @@ -1,11 +1,11 @@
  
  struct MultipleLocks {
diff --git a/patches/struct_000_Optimal.patch b/patches/struct_000_Optimal.patch
index a9048e9..7bcfe65 100644
--- a/patches/struct_000_Optimal.patch
+++ b/patches/struct_000_Optimal.patch
@@ -1,7 +1,7 @@
-diff --git var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpe_2dohl9.cpp var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmppl665vqz.cpp
+diff --git tmp/tmp8gwh1itb.cpp tmp/tmpnm55s8mb.cpp
 index eb39cf3..2562cbc 100644
---- var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpe_2dohl9.cpp
-+++ var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmppl665vqz.cpp
+--- tmp/tmp8gwh1itb.cpp
++++ tmp/tmpnm55s8mb.cpp
 @@ -1,8 +1,8 @@
  
              struct Optimal {
diff --git a/patches/struct_000_Outer.patch b/patches/struct_000_Outer.patch
index e69de29..7f77264 100644
--- a/patches/struct_000_Outer.patch
+++ b/patches/struct_000_Outer.patch
@@ -0,0 +1,21 @@
+diff --git tmp/tmpeanl3uqc.cpp tmp/tmpp_lodp8z.cpp
+index 378f479..e65c5d2 100644
+--- tmp/tmpeanl3uqc.cpp
++++ tmp/tmpp_lodp8z.cpp
+@@ -1,13 +1,13 @@
+ 
+ struct Outer {
++    Inner<double> double_inner;
++Inner<int> int_inner;
+     template<typename T>
+     struct Inner {
+         T value;
+         int id;
+     };
+     
+-    Inner<int> int_inner;
+-    Inner<double> double_inner;
+-};
++    };
+ 
+ Outer outer;
diff --git a/patches/struct_000_Partial.patch b/patches/struct_000_Partial.patch
index 2974e86..dd4f62a 100644
--- a/patches/struct_000_Partial.patch
+++ b/patches/struct_000_Partial.patch
@@ -1,9 +1,9 @@
-diff --git var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpce3q1abn.cpp var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp9i9j4hv6.cpp
-index 2a029c6..eea5c4e 100644
---- var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpce3q1abn.cpp
-+++ var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp9i9j4hv6.cpp
+diff --git tmp/tmplkwy2_fp.cpp tmp/tmp02tn7ire.cpp
+index 2a029c6..56d50cf 100644
+--- tmp/tmplkwy2_fp.cpp
++++ tmp/tmp02tn7ire.cpp
 @@ -1 +1 @@
 -struct Partial { char free1; int size; char* buffer; char free2; Partial(int s) : size(s), buffer(new char[size]), free1('a'), free2('b') {} }; int main() { Partial p(50); return 0; }
 \ No newline at end of file
-+struct Partial { int size; char* buffer; char free1; char free2; Partial(int s) : size(s), buffer(new char[size]), free1('a'), free2('b') {} }; int main() { Partial p(50); return 0; }
++struct Partial { char* buffer; int size; char free1; char free2; Partial(int s) : size(s), buffer(new char[size]), free1('a'), free2('b') {} }; int main() { Partial p(50); return 0; }
 \ No newline at end of file
diff --git a/patches/struct_000_Ptrs.patch b/patches/struct_000_Ptrs.patch
index 69074eb..4603f7d 100644
--- a/patches/struct_000_Ptrs.patch
+++ b/patches/struct_000_Ptrs.patch
@@ -1,7 +1,7 @@
-diff --git var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpt5y66e3l.cpp var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpynogku_1.cpp
+diff --git tmp/tmp5tai3s34.cpp tmp/tmpwgykxo20.cpp
 index 6134502..cb99dcd 100644
---- var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpt5y66e3l.cpp
-+++ var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpynogku_1.cpp
+--- tmp/tmp5tai3s34.cpp
++++ tmp/tmpwgykxo20.cpp
 @@ -1 +1 @@
 -struct Ptrs { char a; void* b; char c; void* d; }; int main() { Ptrs p; return 0; }
 \ No newline at end of file
diff --git a/patches/struct_000_S.patch b/patches/struct_000_S.patch
index dbfd6a6..a82d8d0 100644
--- a/patches/struct_000_S.patch
+++ b/patches/struct_000_S.patch
@@ -1,7 +1,7 @@
-diff --git var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpasndypbl.cpp var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpj_p7uf8a.cpp
+diff --git tmp/tmphprwzdko.cpp tmp/tmpvwpv84zb.cpp
 index 1a1218f..f4e7afb 100644
---- var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpasndypbl.cpp
-+++ var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpj_p7uf8a.cpp
+--- tmp/tmphprwzdko.cpp
++++ tmp/tmpvwpv84zb.cpp
 @@ -1 +1 @@
 -struct S { char a; int b; }; union U { S s; long l; }; int main() { U u; return 0; }
 \ No newline at end of file
diff --git a/patches/struct_000_SingleLock.patch b/patches/struct_000_SingleLock.patch
index 74804e2..5da366d 100644
--- a/patches/struct_000_SingleLock.patch
+++ b/patches/struct_000_SingleLock.patch
@@ -1,7 +1,7 @@
-diff --git var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmppp7m3qjz.cpp var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmprt9fivmt.cpp
+diff --git tmp/tmperjwiusz.cpp tmp/tmpj9hcd8eo.cpp
 index 7ac341a..5ed5fc3 100644
---- var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmppp7m3qjz.cpp
-+++ var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmprt9fivmt.cpp
+--- tmp/tmperjwiusz.cpp
++++ tmp/tmpj9hcd8eo.cpp
 @@ -1,9 +1,9 @@
  
  struct SingleLock {
diff --git a/patches/struct_001_B.patch b/patches/struct_001_B.patch
index 00868fa..4e700d2 100644
--- a/patches/struct_001_B.patch
+++ b/patches/struct_001_B.patch
@@ -1,7 +1,7 @@
-diff --git var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpk1kql60p.cpp var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp84ibi3hk.cpp
+diff --git tmp/tmpmiul1v_s.cpp tmp/tmp3zbdolbp.cpp
 index 37a9bc1..7e51446 100644
---- var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpk1kql60p.cpp
-+++ var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp84ibi3hk.cpp
+--- tmp/tmpmiul1v_s.cpp
++++ tmp/tmp3zbdolbp.cpp
 @@ -1 +1 @@
 -struct B { char x; int y; }; int main() { B b; return 0; }
 \ No newline at end of file
diff --git a/patches/struct_001_BeforeRegion.patch b/patches/struct_001_BeforeRegion.patch
index d497c44..4594ebc 100644
--- a/patches/struct_001_BeforeRegion.patch
+++ b/patches/struct_001_BeforeRegion.patch
@@ -1,7 +1,7 @@
-diff --git var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp1wqhhp_2.cpp var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp_ndxdzok.cpp
+diff --git tmp/tmpgra_nz0t.cpp tmp/tmpfkmz8ij0.cpp
 index 8bc95d4..36a3211 100644
---- var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp1wqhhp_2.cpp
-+++ var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp_ndxdzok.cpp
+--- tmp/tmpgra_nz0t.cpp
++++ tmp/tmpfkmz8ij0.cpp
 @@ -1,7 +1,7 @@
  
  struct BeforeRegion {
diff --git a/patches/struct_001_Derived.patch b/patches/struct_001_Derived.patch
index 14f2231..e69de29 100644
--- a/patches/struct_001_Derived.patch
+++ b/patches/struct_001_Derived.patch
@@ -1,16 +0,0 @@
-diff --git var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpjh2tx0wz.cpp var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp7yva_0i2.cpp
-index 5983060..4d7897c 100644
---- var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpjh2tx0wz.cpp
-+++ var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp7yva_0i2.cpp
-@@ -4,8 +4,8 @@
-                 int b;
-             };
-             struct Derived : Base {
--                char c;
-                 double d;
--            };
-+            char c;
-+                };
-             int main() { Derived d; return 0; }
-             
-\ No newline at end of file
diff --git a/patches/struct_001_FirstBefore.patch b/patches/struct_001_FirstBefore.patch
index 5638ff8..98c3c6d 100644
--- a/patches/struct_001_FirstBefore.patch
+++ b/patches/struct_001_FirstBefore.patch
@@ -1,7 +1,7 @@
-diff --git var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmps6e_cgd4.cpp var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpift02nyp.cpp
+diff --git tmp/tmpmrt16989.cpp tmp/tmpevnkgyos.cpp
 index 24427e0..de4c53b 100644
---- var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmps6e_cgd4.cpp
-+++ var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpift02nyp.cpp
+--- tmp/tmpmrt16989.cpp
++++ tmp/tmpevnkgyos.cpp
 @@ -1,7 +1,7 @@
  
  struct FirstBefore {
diff --git a/patches/struct_001_NormalStruct.patch b/patches/struct_001_NormalStruct.patch
index 58ac620..474a006 100644
--- a/patches/struct_001_NormalStruct.patch
+++ b/patches/struct_001_NormalStruct.patch
@@ -1,7 +1,7 @@
-diff --git var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp_5o3ijl9.cpp var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpv5vzwxno.cpp
+diff --git tmp/tmpg48_ytkc.cpp tmp/tmpdgh5h9os.cpp
 index 6d0f0b6..d0ae289 100644
---- var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp_5o3ijl9.cpp
-+++ var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpv5vzwxno.cpp
+--- tmp/tmpg48_ytkc.cpp
++++ tmp/tmpdgh5h9os.cpp
 @@ -7,8 +7,8 @@ struct IgnoredStruct {
  };
  
diff --git a/patches/struct_001_Outer.patch b/patches/struct_001_Outer.patch
index 8e6d542..24647d0 100644
--- a/patches/struct_001_Outer.patch
+++ b/patches/struct_001_Outer.patch
@@ -1,7 +1,7 @@
-diff --git var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp5bvudjw6.cpp var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpwjv_tzm_.cpp
+diff --git tmp/tmpw6ked_v7.cpp tmp/tmplnn1ca3o.cpp
 index b83699b..a46c3d8 100644
---- var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp5bvudjw6.cpp
-+++ var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpwjv_tzm_.cpp
+--- tmp/tmpw6ked_v7.cpp
++++ tmp/tmplnn1ca3o.cpp
 @@ -1 +1 @@
 -struct Inner { char x; int y; }; struct Outer { char a; Inner b; char c; }; int main() { Outer o; return 0; }
 \ No newline at end of file
diff --git a/patches/struct_001_ProcessedStruct.patch b/patches/struct_001_ProcessedStruct.patch
index 5881469..17e19a6 100644
--- a/patches/struct_001_ProcessedStruct.patch
+++ b/patches/struct_001_ProcessedStruct.patch
@@ -1,7 +1,7 @@
-diff --git var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpok9ceg9t.cpp var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpybdf7ee5.cpp
+diff --git tmp/tmp6ou58cdz.cpp tmp/tmp5r7wf61m.cpp
 index 420b9bb..c9ddb27 100644
---- var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpok9ceg9t.cpp
-+++ var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpybdf7ee5.cpp
+--- tmp/tmp6ou58cdz.cpp
++++ tmp/tmp5r7wf61m.cpp
 @@ -10,9 +10,9 @@
                  int b;
              };
diff --git a/patches/struct_001_SecondStruct.patch b/patches/struct_001_SecondStruct.patch
index 92d33c1..bd6f758 100644
--- a/patches/struct_001_SecondStruct.patch
+++ b/patches/struct_001_SecondStruct.patch
@@ -1,7 +1,7 @@
-diff --git var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp4n7i2hr3.cpp var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpg97e9l1_.cpp
+diff --git tmp/tmppv0qvfdd.cpp tmp/tmpohmfboq_.cpp
 index cb6de31..9b7833f 100644
---- var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp4n7i2hr3.cpp
-+++ var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpg97e9l1_.cpp
+--- tmp/tmppv0qvfdd.cpp
++++ tmp/tmpohmfboq_.cpp
 @@ -7,8 +7,8 @@ struct FirstStruct {
  };
  
diff --git a/patches/struct_002_FirstInRegion.patch b/patches/struct_002_FirstInRegion.patch
index 4f4fe53..8022163 100644
--- a/patches/struct_002_FirstInRegion.patch
+++ b/patches/struct_002_FirstInRegion.patch
@@ -1,7 +1,7 @@
-diff --git var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpur5ueifl.cpp var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpbvx6pmvf.cpp
+diff --git tmp/tmpzn555gz_.cpp tmp/tmpmb0olksc.cpp
 index 9fbcf4a..82836ea 100644
---- var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpur5ueifl.cpp
-+++ var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpbvx6pmvf.cpp
+--- tmp/tmpzn555gz_.cpp
++++ tmp/tmpmb0olksc.cpp
 @@ -7,8 +7,8 @@ struct BeforeRegion {
  
  // paddington-off
diff --git a/patches/struct_002_InOffRegion.patch b/patches/struct_002_InOffRegion.patch
index f691165..04a80e1 100644
--- a/patches/struct_002_InOffRegion.patch
+++ b/patches/struct_002_InOffRegion.patch
@@ -1,7 +1,7 @@
-diff --git var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp9lxgf98r.cpp var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpyn79f6on.cpp
+diff --git tmp/tmpa_jwajaq.cpp tmp/tmpw5t3_vi6.cpp
 index 51bb82d..9b592e8 100644
---- var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp9lxgf98r.cpp
-+++ var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpyn79f6on.cpp
+--- tmp/tmpa_jwajaq.cpp
++++ tmp/tmpw5t3_vi6.cpp
 @@ -7,8 +7,8 @@ struct BeforeRegion {
  
  // paddington-off
diff --git a/patches/struct_002_InRegion.patch b/patches/struct_002_InRegion.patch
index cd688cd..5c3ce85 100644
--- a/patches/struct_002_InRegion.patch
+++ b/patches/struct_002_InRegion.patch
@@ -1,7 +1,7 @@
-diff --git var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmphrqpnljh.cpp var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpzwpy4zbl.cpp
+diff --git tmp/tmp49p7ylnt.cpp tmp/tmplg8vmxh3.cpp
 index 24427e0..2c8d828 100644
---- var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmphrqpnljh.cpp
-+++ var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpzwpy4zbl.cpp
+--- tmp/tmp49p7ylnt.cpp
++++ tmp/tmplg8vmxh3.cpp
 @@ -13,8 +13,8 @@ struct SecondBefore {
  
  // paddington-off
diff --git a/patches/struct_002_InnerRegion.patch b/patches/struct_002_InnerRegion.patch
index 25d7931..cef8bb9 100644
--- a/patches/struct_002_InnerRegion.patch
+++ b/patches/struct_002_InnerRegion.patch
@@ -1,7 +1,7 @@
-diff --git var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpq27cekjx.cpp var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpex0z2c15.cpp
+diff --git tmp/tmpji4rmvui.cpp tmp/tmponmihy1x.cpp
 index 8bc95d4..56e1032 100644
---- var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpq27cekjx.cpp
-+++ var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpex0z2c15.cpp
+--- tmp/tmpji4rmvui.cpp
++++ tmp/tmponmihy1x.cpp
 @@ -14,8 +14,8 @@ struct OuterRegion {
  
  // paddington-off
diff --git a/patches/struct_002_SecondIgnored.patch b/patches/struct_002_SecondIgnored.patch
index 2260d6f..1bf1a6b 100644
--- a/patches/struct_002_SecondIgnored.patch
+++ b/patches/struct_002_SecondIgnored.patch
@@ -1,7 +1,7 @@
-diff --git var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpwad49mzh.cpp var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmphyc6qhcu.cpp
+diff --git tmp/tmp7rgls_5s.cpp tmp/tmp_iekoga8.cpp
 index 1d95166..1bf56bd 100644
---- var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpwad49mzh.cpp
-+++ var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmphyc6qhcu.cpp
+--- tmp/tmp7rgls_5s.cpp
++++ tmp/tmp_iekoga8.cpp
 @@ -10,9 +10,9 @@
              };
              // paddington-ignore
diff --git a/patches/struct_002_SingleLineIgnored.patch b/patches/struct_002_SingleLineIgnored.patch
index 1661696..5e65a12 100644
--- a/patches/struct_002_SingleLineIgnored.patch
+++ b/patches/struct_002_SingleLineIgnored.patch
@@ -1,7 +1,7 @@
-diff --git var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpj4c6du25.cpp var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp51oo5md_.cpp
+diff --git tmp/tmphtqeqh6a.cpp tmp/tmp2xmvl50b.cpp
 index 4c04dff..3f3b57c 100644
---- var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpj4c6du25.cpp
-+++ var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp51oo5md_.cpp
+--- tmp/tmphtqeqh6a.cpp
++++ tmp/tmp2xmvl50b.cpp
 @@ -1,9 +1,9 @@
  
              // paddington-ignore
diff --git a/patches/struct_002_UpperCaseIgnored.patch b/patches/struct_002_UpperCaseIgnored.patch
index a1a35e2..e6d93c2 100644
--- a/patches/struct_002_UpperCaseIgnored.patch
+++ b/patches/struct_002_UpperCaseIgnored.patch
@@ -1,7 +1,7 @@
-diff --git var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpho9qkj8x.cpp var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpzazps2r0.cpp
+diff --git tmp/tmpwfb37m2a.cpp tmp/tmpr_qwegrw.cpp
 index 742238f..55faba6 100644
---- var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpho9qkj8x.cpp
-+++ var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpzazps2r0.cpp
+--- tmp/tmpwfb37m2a.cpp
++++ tmp/tmpr_qwegrw.cpp
 @@ -1,9 +1,9 @@
  
              // PADDINGTON-IGNORE
diff --git a/patches/struct_002_WhitespaceIgnored.patch b/patches/struct_002_WhitespaceIgnored.patch
index 5487227..4aeb7df 100644
--- a/patches/struct_002_WhitespaceIgnored.patch
+++ b/patches/struct_002_WhitespaceIgnored.patch
@@ -1,7 +1,7 @@
-diff --git var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp4u27pi8k.cpp var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmplu0uyufb.cpp
+diff --git tmp/tmpjq9ybxdq.cpp tmp/tmp7v89n2px.cpp
 index 420b9bb..f934305 100644
---- var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp4u27pi8k.cpp
-+++ var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmplu0uyufb.cpp
+--- tmp/tmpjq9ybxdq.cpp
++++ tmp/tmp7v89n2px.cpp
 @@ -1,9 +1,9 @@
  
              //   paddington-ignore   
diff --git a/patches/struct_003_OuterRegion.patch b/patches/struct_003_OuterRegion.patch
index 4ab5f9b..8db9bcf 100644
--- a/patches/struct_003_OuterRegion.patch
+++ b/patches/struct_003_OuterRegion.patch
@@ -1,7 +1,7 @@
-diff --git var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpsyswjrpb.cpp var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpr1555m27.cpp
+diff --git tmp/tmpt2f139n9.cpp tmp/tmp0bekmqfc.cpp
 index 8bc95d4..3897bf3 100644
---- var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpsyswjrpb.cpp
-+++ var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpr1555m27.cpp
+--- tmp/tmpt2f139n9.cpp
++++ tmp/tmp0bekmqfc.cpp
 @@ -7,8 +7,8 @@ struct BeforeRegion {
  
  // paddington-off
diff --git a/patches/struct_003_SecondAfter.patch b/patches/struct_003_SecondAfter.patch
index d440396..1f568b8 100644
--- a/patches/struct_003_SecondAfter.patch
+++ b/patches/struct_003_SecondAfter.patch
@@ -1,7 +1,7 @@
-diff --git var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmplk_n6ruk.cpp var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp1e71rl5h.cpp
+diff --git tmp/tmp81o6u2np.cpp tmp/tmpufjxhap8.cpp
 index 24427e0..82b56c4 100644
---- var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmplk_n6ruk.cpp
-+++ var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp1e71rl5h.cpp
+--- tmp/tmp81o6u2np.cpp
++++ tmp/tmpufjxhap8.cpp
 @@ -26,8 +26,8 @@ struct FirstAfter {
  };
  
diff --git a/patches/struct_003_SecondInRegion.patch b/patches/struct_003_SecondInRegion.patch
index cea0a59..3321bec 100644
--- a/patches/struct_003_SecondInRegion.patch
+++ b/patches/struct_003_SecondInRegion.patch
@@ -1,7 +1,7 @@
-diff --git var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp3otkkz_j.cpp var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpx9k9suas.cpp
+diff --git tmp/tmpis5pk8us.cpp tmp/tmpb9jvjjte.cpp
 index 9fbcf4a..c279c38 100644
---- var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp3otkkz_j.cpp
-+++ var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmpx9k9suas.cpp
+--- tmp/tmpis5pk8us.cpp
++++ tmp/tmpb9jvjjte.cpp
 @@ -13,8 +13,8 @@ struct FirstInRegion {
  };
  
diff --git a/patches/struct_004_SecondBefore.patch b/patches/struct_004_SecondBefore.patch
index b7fd9cf..0890853 100644
--- a/patches/struct_004_SecondBefore.patch
+++ b/patches/struct_004_SecondBefore.patch
@@ -1,7 +1,7 @@
-diff --git var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp_yf3g3my.cpp var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp7nxzxz7s.cpp
+diff --git tmp/tmpqo2s_gf1.cpp tmp/tmpqym7u1ru.cpp
 index 24427e0..1ea8176 100644
---- var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp_yf3g3my.cpp
-+++ var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp7nxzxz7s.cpp
+--- tmp/tmpqo2s_gf1.cpp
++++ tmp/tmpqym7u1ru.cpp
 @@ -6,8 +6,8 @@ struct FirstBefore {
  };
  
diff --git a/patches/struct_004_StillInOuter.patch b/patches/struct_004_StillInOuter.patch
index facc778..a2a0312 100644
--- a/patches/struct_004_StillInOuter.patch
+++ b/patches/struct_004_StillInOuter.patch
@@ -1,7 +1,7 @@
-diff --git var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmptf_1g5ex.cpp var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp5tkauz58.cpp
+diff --git tmp/tmprukxt973.cpp tmp/tmpen19yjsn.cpp
 index 8bc95d4..c4c94ec 100644
---- var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmptf_1g5ex.cpp
-+++ var/folders/vn/llg2yrpd6q17y53ny2f6yfyw0000gq/T/tmp5tkauz58.cpp
+--- tmp/tmprukxt973.cpp
++++ tmp/tmpen19yjsn.cpp
 @@ -21,8 +21,8 @@ struct InnerRegion {
  // paddington-on
  
-- 
2.40.1

